### PR TITLE
prl-tools: add patch against 5.19

### PR DESCRIPTION
**Pull request by wegank**
###### Description of changes

~TODO: Test if it builds on 5.19 when the kernel arrives at nixos-unstable.~ Builds well on 5.19.

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by wegank** [comment]
@ofborg build linuxKernel.packages.linux_5_19.prl-tools

**Review by catap — APPROVED** [review]

File: pkgs/os-specific/linux/prl-tools/default.nix [hunk 1]
```diff
@@ -39,7 +56,10 @@ stdenv.mkDerivation rec {
     fi
   '';
 
-  patches = lib.optionals (lib.versionAtLeast kernel.version "5.18") [ ./prl-tools.patch ];
+  patches = lib.concatMap
```
Comment (SuperSandro2000) [inline pkgs/os-specific/linux/prl-tools/default.nix:59]: This is pretty short written out so I would prefer that. Also it makes it easier to add more patches later on.

Comment (wegank) [reply pkgs/os-specific/linux/prl-tools/default.nix]: Done.

File: pkgs/os-specific/linux/prl-tools/default.nix [hunk 2]
```diff
@@ -39,7 +56,8 @@ stdenv.mkDerivation rec {
     fi
   '';
 
-  patches = lib.optionals (lib.versionAtLeast kernel.version "5.18") [ ./prl-tools.patch ];
+  patches = lib.optional (lib.versionAtLeast kernel.version "5.18") ./prl-tools-5.18.patch
+    ++ lib.optional (lib.versionAtLeast kernel.version "5.19") ./prl-tools-5.19.patch;
```
Comment (catap) [inline pkgs/os-specific/linux/prl-tools/default.nix:60]: `lib.optionals`?

Comment (wegank) [reply pkgs/os-specific/linux/prl-tools/default.nix:60]: `lib.optional` is of type `bool -> a -> [a]`, so it should work. Is `lib.optionals` preferred in nixpkgs?

Comment (catap) [reply pkgs/os-specific/linux/prl-tools/default.nix:60]: let ask @SuperSandro2000 but I think that yes.

Comment (SuperSandro2000) [reply pkgs/os-specific/linux/prl-tools/default.nix:60]: Does not really matter as long as we don't end up with nested lists

**Comment by SuperSandro2000** [comment]
@ofborg build linuxPackages.prl-tools

**Review by catap — APPROVED** [review]

**Comment by wegank** [comment]
Only if the package is free...

**Pull request merged**